engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -100,6 +100,8 @@ def verify_extension_activation(product)
               'The product is not available for this instance'
             )
           end
+          logger.info "Product #{@product.product_string} available for this instance"
+          InstanceVerification.update_cache(request.remote_ip, @system.login, @product.id)
```

Comment:
Isn't this `product` instead of `@product`?

```suggestion
          InstanceVerification.update_cache(request.remote_ip, @system.login, product.id)
```